rust/processor/src/db/common/models/emojicoin_models/models/arena_candlestick.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+    db::common::models::emojicoin_models::{enums::Period, json_types::SwapEvent},
+    schema::arena_candlestick,
+};
+use bigdecimal::{BigDecimal, Zero};
+use chrono::{DurationRound, NaiveDateTime};
+use field_count::FieldCount;
+use num::FromPrimitive;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub struct Prices {
+    pub open_price: BigDecimal,
+    pub high_price: BigDecimal,
+    pub low_price: BigDecimal,
+    pub close_price: BigDecimal,
+
+    pub open_timestamp: (i64, i64),
+    pub close_timestamp: (i64, i64),
+}
+
+#[derive(Clone)]
+pub struct ArenaCandlestickDiffModelBuilder {
+    pub melee_id: BigDecimal,
+
+    pub period: Period,
+    pub start_time: NaiveDateTime,
+
+    pub prices: Option<Prices>,
+
+    pub volume: BigDecimal,
+    pub integrator_fees: BigDecimal,
+    pub n_swaps: BigDecimal,
+}
+
+impl ArenaCandlestickDiffModelBuilder {
+    pub fn merge(sticks: Vec<Self>) -> Vec<Self> {
+        let mut sticks_map: HashMap<(BigDecimal, Period, NaiveDateTime), Self> = HashMap::new();
+
+        for stick in sticks {
+            let stick_clone = stick.clone();
+            sticks_map
+                .entry((
+                    stick.melee_id.clone(),
+                    stick.period.clone(),
+                    stick.start_time.clone(),
+                ))
+                .and_modify(|s| {
```

Comment:
```suggestion
        for incoming in sticks {
            let incoming_clone = incoming.clone();
            sticks_map
                .entry((
                    incoming.melee_id.clone(),
                    incoming.period.clone(),
                    incoming.start_time.clone(),
                ))
                .and_modify(|current| {
```

This is really difficult to read without meaningful variable names, it uses `s.prices()`, `s_prices`, `stick.prices`, and `stick_prices`

Reply:
You already fixed it in another PR before I even mentioned it!! 😀